websites/S/Speedtest/presence.ts

```diff
@@ -0,0 +1,93 @@
+const presence = new Presence({
+    clientId: "817385570912174121"
+  }),
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "image"
+  }, showISP = await presence.getSetting("showISP");
+
+  if (window.location.hostname.includes("speedtest")) {
+     if (window.location.pathname.includes("/run")) {
+      const server = document.querySelector(".result-item-icon.result-item-host .result-data, .result-item-icon.result-item-host .result-label").textContent + " - " + document.querySelector(".result-item-icon.result-item-host .result-data, .result-item-icon.result-item-host .result-label .result-name").textContent,
+      ping = document.querySelector(".result-item-ping .result-data").textContent + document.querySelector(".result-item-ping .result-data-unit").textContent,
+      download = document.querySelector(".result-item-download .result-data").textContent + document.querySelector(".result-item-download .result-data-unit").textContent,
+      upload = document.querySelector(".result-item-upload .result-data").textContent + document.querySelector(".result-item-upload .result-data-unit").textContent;
+      let isp = document.querySelector(".result-item-icon.result-item-isp .result-label").textContent;
+      if (!showISP) isp = "Hidden";
+      // 27 & 29 = nothing + unit
+      if (upload.length == 29) {
+        presenceData.details = `Live results - Ping ${ping.length == 27 ? "Testing" : ping} | Download ${download.length == 29 ? "Testing" : download} | Upload ${upload.length == 29 ? "Testing" : upload}`;
+        presenceData.state = `ISP: ${isp} | Server: ${server}`;
```

Comment:
could use a ternary operator for this to possibly make it show only `Server: {}` when their ISP is hidden.